ansible/roles/zip-merge-script-to-puhti/files/update_zip_with_sources.py

```diff
@@ -18,49 +18,79 @@ def update_zip_with_tars(zipfilename, tarfilenames):
                 # paths should be autogenerated in the zip
                 continue
             iobuf = tar.extractfile(tar_item)
-            zipfile.writestr(tar_item.name, iobuf.read())
-        tar.close()
-    zipfile.close()
+            if pathlib.Path(tar_item.name).suffix in args.no_compress:
+                zip_fileobj.writestr(
+                    tar_item.name,
+                    iobuf.read(),
+                    compress_type=ZIP_STORED,
+                )
+            else:
+                zip_fileobj.writestr(tar_item.name, iobuf.read())
+
+
+def update_zip_with_targets(args):
+    """
+    Create or update the zip in args.zipfilename with each target from args.sourcepaths.
+
+    Targets are archives of some kind, and the paths within the targets will be used for
+    storing the files in the zip. The only target file type currently supported is tar.
+    In the future, other archives, like zips, should also be processed for contents.
+
+    Files having an extension in args.no_compress will be stored without compression.
+    """
+    with ZipFile(args.zipfilename, mode="a", compression=ZIP_DEFLATED) as zipfile:
+        for sourcepath in args.sourcepaths:
+            if sourcepath.suffix == ".tar":
+                update_zip_with_tar(zipfile, sourcepath, args)
+            else:
+                print(
+                    f"Warning: tried to update zip with unknown format {sourcepath}",
+                    sys.stderr,
+                )
 
 
 if __name__ == "__main__":
-    import os
-    import sys
     import argparse
 
     argparser = argparse.ArgumentParser(
         prog="collect-to-zip",
         epilog="Optionally write additional source filenames to stdin",
     )
-    argparser.add_argument("zipfilename", help="The target zip")
-    argparser.add_argument("--dir", type=str, help="Scan directory for sources")
+    argparser.add_argument("zipfilename", type=pathlib.Path, help="The target zip")
+    argparser.add_argument(
+        "--dir", type=pathlib.Path, help="Scan directory for sources"
+    )
     argparser.add_argument(
         "--sourcelist", type=str, help="Read file for sources, or - for stdin"
     )
     argparser.add_argument("--verbose", action="store_true")
+    argparser.add_argument(
+        "--no-compress",
+        "-n",
+        help="A colon-separated list of suffixes to not compress, as in zip",
+        default=".jpg:.jpeg",
```

Comment:
We need to add jp2, that's the extension NLF actually uses.

Reply:
Can add either in the Airflow code or here, but I'll add it here now.